src/mame/machine/mpu4_characteriser_pal.h

```diff
@@ -0,0 +1,1432 @@
+	//                               "RGT 1.0"
+	// Magnificent 7s                "MA7 1.6"
+	//                               "MAS 1.3"
+	// Make A Mint                   "MAM 0.4"
+	//                               "MMG 0.5"
+	// Pot Black                     "PBS 0.6"
+	//                               "PBG 1.6"
+	// Place Your Bets               "PYB 0.7"
+	//                               "PYH 0.6"
+	// Cloud Nine                    "CT3 0.2"
+	//                               "CT2 0.2"
+	//                               "CT5 0.2"
+	// Tutti Fruity                  "F1U 0.1"
+	//                               "F2U 0.1"
+	//                               "F3U 0.1"
+	// Cash Attack                   "CSA 1.2"
+	//                               "CAA 2.3"
+	// Red Hot Roll                  "HHN 0.2 / CR4 0.2"
+	//                               "CLD 0.3 / CRU 0.3"
+	//                               "HHN 0.3 / CR4 0.3"
+	//                               "RRD 0.3 / CR4 0.3"
+	// Up Up and Away                "UPS 2.2"
+	//                               "UUA 2.2"
+	// Hi Jinx                       "JNX 1.0"
+	//                               "JNS 0.3"
+	// Cash Lines                    "CLS 0.4"
+	//                               "CLI 1.2"
+	// Lucky Strike                  "LSS 0.7"
+	//                               "LST 1.0"
+	// 10 X 10                       "T2T 0.1"
+	//                               "T3T 0.1"
+	//                               "TST 0.1"
+	// Boulder Dash                  "BLD 1.0"
+	//                               "BLS 0.2"
+	// Hot Rod                       "ROD 0.4"
+	// Buccaneer                     "BUG 0.5"
+	//                               "BUS 0.2"
+	// Bagatelle                     "EL1 0.1"
+	//                               "EL2 0.1"
+	//                               "EL3 1.0"
+	// The Crystal Dome              "CD2 1.2"
+	//                               "CD2 1.0"
+	//                               "CD2 0.2"
+	// Cash Machine                  "CMA 0.8"
+	//                               "CMH 0.7"
+	// Top Tenner                    "TTS 0.4"
+	//                               "TTH 1.1"
+	//                               "TTH 1.2"
+	// Ten Out Of Ten                "TOC 0.4"
+	//                               "TOT 0.6"
+	// Everyone's A Winner           "ENN 0.1 / ER4 0.1"
+	//                               "EON 0.1 / ER4 0.1"
+	//                               "EUN 0.1 / ER4 0.1"
+	// Let The Good Times Roll       "GTR 2.0"
+	//                               "GTS 1.0"
+	//                               "GTS 0.2"
+	//                               "GTR 1.1"
+	// Super Streak                  "CS4 0.2 / CSP 0.2"
+	//                               "CS4 0.2 / EEH 0.2"
+	//                               "CS4 0.2 / STC 0.2"
+	//                               "STC 0.1"
+	static constexpr uint8_t m4dtri98_characteriser_prot[8] = { 0x03, 0xAF, 0x87, 0xAB, 0xA3, 0x8F, 0x87, 0x83 };
```

Comment:
What’s the ultimate plan for these things?  I’m a bit concerned about putting them all in the header.  The PAL content is really game data.  Also, having them all in a header like this increases fragility, causing rebuilds of the drivers when any of them change.

Reply:
for now I still need to study them further, and look for patterns.

ultimately the PALs are going to be some kind of equations, my hope is that by grouping information on them I'll be able to spot a weakness or similar.  I decided on the header because

a) many are used in different places (it seems they were taken from a pool / recycled)
b) it gives me a single place to look for a list of versions using the same scramble / patterns (which makes checking them easier)
c) it gives me a single place to look when studying the patterns.
d) it gives a good indication of which chips were reused more frequently than others
e) if somebody is repairing these things, again it gives a good idea of where they might find a compatible part

where they end up in the end I'm not sure, I don't really like the idea of fake device ROMs for them, because the pass sequences / lamp sequences have little to do with the actual internal data and half the mess from before was people trying to do this and creating multiple incomplete / inaccurate files to hack around the protections, which in turn had even resulted in people creating layouts that were designed around incorrect lamp scrambles.

I concede that it isn't ideal, but the previous situation was unworkable, and this allows for progress to be made.